code/ai/aigoals.cpp

```diff
@@ -1022,17 +1022,30 @@ void ai_add_goal_sub_sexp( int sexp, int type, ai_info *aip, ai_goal *aigp, char
 		aigp->ai_mode = AI_GOAL_KEEP_SAFE_DISTANCE;
 		break;
 
+	case OP_AI_FLY_TO_SHIP:
 	case OP_AI_STAY_NEAR_SHIP:
 	{
 		bool is_nan, is_nan_forever;
 
 		aigp->target_name = ai_get_goal_target_name( CTEXT(CDR(node)), &aigp->target_name_index );
 		aigp->priority = atoi( CTEXT(CDDR(node)) );
+
+		// distance from ship
 		if ( CDDDR(node) < 0 )
 			aigp->float_data = 300.0f;
 		else
 			aigp->float_data = i2fl(eval_num(CDDDR(node), is_nan, is_nan_forever));
-		aigp->ai_mode = AI_GOAL_STAY_NEAR_SHIP;
+
+		// (the next argument is whether to afterburn)
```

Comment:
Put a brief additional comment here saying that it's handled further below, instead of right here (Not that I'm actually sure handling the flag in one fell swoop at the end of the function is a good way to do it...).